app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        for arguments, result in cached_results.items():
+            if arguments == args:
+                print("Getting from cache")
+                return result
```

Comment:
Try to rewrite with one return statement like so:
if data not in cache -> add data to cache
...
return data from cache